src/Firebase/firebase.js

```diff
@@ -209,19 +264,59 @@ export const doDeleteInternship = (internship_id) => {
       console.log(error);
     });
 };
-/*export function getImageURL(image) {
-  var storageref = firebase.storage().ref('projectPhoto/' + image);
-var uploadTask = storageref.put(image);
-
-uploadTask.on('state_changed', function(snapshot){
-}, function(error){
-  console.error(error);
-}, function() {
-  uploadTask.snapshot.ref.getDownloadURL().then(function(downloadURL) {
-    console.log('File available at', downloadURL);
-    image = downloadURL;
-    
+
+
+export const doEditProject = async (obj,project_id) => {
+  let user = firebase.auth().currentUser;
+  if (!user) {
+    alert("Please login to add a project");
+    return;
+  }
+  let projectRef = firebase.database().ref("projects/" + project_id);
+  projectRef
+  .child("leader_id")
+  .once("value")
+  .then(function (snapshot) {
+    if (snapshot.val() !== user.uid) {
+      return;
+    }
   });
-});
- 
-} */
\ No newline at end of file
+  let photoUrl='';  
+   projectRef.child("photo").once("value").then(await function(snapshot){if(snapshot.val()){photoUrl= snapshot.val()}
+  else{return 'https://images.unsplash.com/photo-1639413665566-2f75adf7b7ca?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxlZGl0b3JpYWwtZmVlZHwyN3x8fGVufDB8fHx8&auto=format&fit=crop&w=500&q=60'}});
+  if(obj.photo!==''){
+ try{
+  let imgDb = await firebase.storage().ref(`projectPhoto/${project_id}`)
+  let newimg= await imgDb.put(obj.photo)
+    photoUrl = await newimg.ref.getDownloadURL()
+   alert()
+  let uid = user.uid;
+  let leaderName = user.displayName;
+  const createdAt = Date.now();
+  console.log(photoUrl)
+projectRef
+  .set( {
+    name: obj.title,
+    desc: obj.desc,
+    links: obj.links,
```

Comment:
Hey, small note: in future, since the key and values are basically same as in obj, we could spread the values directly and hence need to enter only the change values. eg: https://github.com/IEDCMEC/iedc-collab-frontend/blob/94d5df6d6799e782687ffaa300b5e6184ba660f7/src/Firebase/firebase.js#L86